src/RichXMLParser.m

```diff
@@ -57,8 +57,15 @@ -(BOOL)parseRichXML:(NSData *)xmlData
         xmlDocument = [[NSXMLDocument alloc] initWithData:xmlData
                                                         options:NSXMLNodeOptionsNone
                                                         error:&error];
-        if (error) {
-            if (error.code == 73) return NO;
+        if (xmlDocument == nil && error != nil) {
+			if ([error.domain isEqualToString:NSXMLParserErrorDomain]) {
+				if (error.code == NSXMLParserGTRequiredError)
+					return NO;
+				if (error.code == NSXMLParserPrematureDocumentEndError)
+					return NO;
+			}
+			// Do we even want to try this?
```

Comment:
On further investigation, I was a little misled by what was happening in Xcode. Unlike in https://github.com/ViennaRSS/vienna-rss/issues/619 it wasn't actually crashing. I put breakpoints before and after the try block, and since it wasn't getting out of the try block, I assumed that execution wasn't continuing in the method. However, what's happening is that there is a series of about 15 assertion failures in -initWithData:. The assertion failures pause the debugger, and they log to the console, but they don't actually crash the app, and once you get through all the assertion failures, execution continues after the try block. So, I think I would withdraw https://github.com/ViennaRSS/vienna-rss/pull/623/commits/4355c40d2cbb27a35beb12eefa3fc72ca6ab9fe2

However, https://github.com/ViennaRSS/vienna-rss/pull/623/commits/a74eb52d98402356b301807200779a910f5f640a is still a good idea.

josh64x2, what do all of those error codes you listed (NSXMLParserNAMERequiredError, etc.) represent?